.github/workflows/build-lint-test.yml

```diff
@@ -0,0 +1,41 @@
+# This workflow will install Python dependencies, run tests and lint on Python 3.11.
+# This initiates on every push or pull_request to develop
+
+name: Build, Lint, and Test
+
+on:
+  push:
+    branches: [ "develop" ]
+  pull_request:
+    branches: [ "develop" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.11
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.11"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Set PYTHONPATH
+      run: echo "PYTHONPATH=${{ github.workspace }}" >> $GITHUB_ENV
```

Comment:
Do we need to set the PYTHONPATH here?  Which bit of the workflow fails if we don't have it?

Reply:
I had the same problem when working on this issue (I didn't create a PR as this was created already). My fix was using `python -m pytest` rather than calling pytest. This meant PYTHONPATH was no longer required as it ensures the tests are in the current python enviroment. Maybe this could be used instead?

Reply:
Thanks very much @Nigel007 - one of the delays in my next review of this PR was that I was trying to figure out why none of our other Python GA workflows needed the PYTHONPATH like this.

I think I'd prefer to go with the solution suggested by @Nigel007 , even though @lourencofsilva 's solution is also plausible, so that the GA workflow set up matches the dev environment set up as closely as possible (where we don't set PYTHONPATH explicitly either).

Reply:
I agree with this, `python -m pytest` is the best way of going about this. Thanks for the input @Nigel007. I have just altered this and removed the PYTHONPATH setup.